go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -235,6 +235,15 @@ func (bsnc *BaseSecondaryNetworkController) ensurePodForSecondaryNetwork(pod *ka
 		return err
 	}
 
+	if bsnc.IsPrimaryNetwork() && (bsnc.TopologyType() == "layer2" || bsnc.TopologyType() == "layer3") {
+		network := &nadapi.NetworkSelectionElement{
+			// TODO: we need to get the NAD name here, not the network name.
+			Name:      bsnc.GetNetworkName(),
```

Comment:
@tssurya I assume your helper will be available here. 

We also need to figure out what's the NAD name here, in order to build the network selection element.